apps/daimo-bot/src/PaymentActionProcessor.ts

```diff
@@ -0,0 +1,222 @@
+import {
+  EAccount,
+  assert,
+  dollarsToAmount,
+  encodeRequestId,
+  formatDaimoLink,
+  generateRequestId,
+} from "@daimo/common";
+import { NeynarAPIClient } from "@neynar/nodejs-sdk";
+import {
+  CONNECT_FC_MESSAGE,
+  PAYMENT_CONNECT_FC_MESSAGE,
+  REQUEST_PAYMENT_MESSAGE,
+} from "./botResponses";
+import { trpcClient } from "./trpcClient";
+import { SendCastOptions, TRPCClient, WebhookEvent } from "./types";
+
+// 4 cases of Payment request:
+
+// Requesting:
+// Case 1: Alice doesn't have FC linked ❌, requests $ from anyone (open-ended post)
+//      Action 1: Daimobot responds with a link to register with Farcaster. Alice registers, then Daimobot responds with a link to request $
+//  Case 1a: Alice doesn't have FC linked ❌ but has ENS linked on FC, requests $ from anyone
+//     Action 1a:  Daimobot responds with link that requests $ from anyone to Alice's ENS
+// Case 2: Alice has FC linked ✅, requests $ from anyone
+//      Action 2: Daimobot responses with link that requests $ from anyone to Alice's Daimo address
+
+// Paying:
+// Case 3: Alice responds to Bobs post to pay him, Bob doesn't have FC linked ❌
+//     Action 3:  Daimobot responds with a link to register with Farcaster. Bob registers, then Daimobot responds with a link to request $
+//  Case 3a: Alice responds to Bobs post to pay him, Bob doesn't have FC linked ❌ but has ENS linked on FC
+//     Action 3a:  Daimobot responds with link that requests $ from anyone to Bob's ENS
+// Case 4: Alice responds to Bobs post to pay him, Bob has FC linked ✅
+//     Action 4: Daimobot responds with link that requests $ from anyone to Bob's Daimo address
+
+assert(!!process.env.NEYNAR_API_KEY, "NEYNAR_API_KEY is not defined");
+const _neynarClient = new NeynarAPIClient(process.env.NEYNAR_API_KEY);
+
+assert(
+  !!process.env.DAIMOBOT_SIGNER_UUID,
+  "DAIMOBOT_SIGNER_UUID is not defined"
+);
+const DAIMOBOT_SIGNER_UUID = process.env.DAIMOBOT_SIGNER_UUID;
+
+export class PaymentActionProcessor {
+  private text: string;
+  private castId: string;
+  private senderFid: number;
+  private authorUsername: string;
+  private parentAuthorFid: number | null;
+  private trpcClient: TRPCClient;
+  private neynarClient: NeynarAPIClient;
+
+  constructor(
+    event: WebhookEvent,
+    trpc: TRPCClient = trpcClient,
+    neynarClient: NeynarAPIClient = _neynarClient
+  ) {
+    const { data } = event;
+    this.text = data.text;
+    this.castId = data.hash;
+    this.authorUsername = data.author.username;
+    this.parentAuthorFid = data.parent_author.fid;
+    this.senderFid = data.author.fid;
+
+    this.trpcClient = trpc;
+    this.neynarClient = neynarClient;
+  }
+
+  async process() {
```

Comment:
recommend putting short summary comments > in a sentence or two, what does `process()` do?